pkg/osbuild/ostree_deploy_container_stage.go

```diff
@@ -0,0 +1,67 @@
+package osbuild
+
+import (
+	"fmt"
+	"regexp"
+)
+
+const ostreeContainerTargetImgrefRegex = "^(ostree-remote-registry|ostree-image-signed|ostree-unverified-registry):.*$"
+
+// Options for the org.osbuild.ostree.deploy.container stage.
+type OSTreeDeployContainerStageOptions struct {
+
+	// Name of the stateroot to be used in the deployment
+	OsName string `json:"osname"`
+
+	// Additional kernel command line options
+	KernelOpts []string `json:"kernel_opts,omitempty"`
+
+	// Image ref used as the source of truth for updates
+	TargetImgref string `json:"target_imgref"`
+
+	// Identifier to locate the root file system (uuid or label)
+	Rootfs *Rootfs `json:"rootfs,omitempty"`
+
+	// Mount points of the final file system
+	Mounts []string `json:"mounts,omitempty"`
+}
+
+func (OSTreeDeployContainerStageOptions) isStageOptions() {}
+
+func (options OSTreeDeployContainerStageOptions) validate() error {
+	exp := regexp.MustCompile(ostreeContainerTargetImgrefRegex)
```

Comment:
(nitpick) Maybe it's worth moving this out of the function and into the global context. The advantage is that any regex error becomes an error at `go test` time (and it's compiled onlyonce) Something like:
```go
diff --git a/pkg/osbuild/ostree_deploy_container_stage.go b/pkg/osbuild/ostree_deploy_container_stage.go
index 78e01f6af..f5a36f455 100644
--- a/pkg/osbuild/ostree_deploy_container_stage.go
+++ b/pkg/osbuild/ostree_deploy_container_stage.go
@@ -5,7 +5,7 @@ import (
        "regexp"
 )
 
-const ostreeContainerTargetImgrefRegex = "^(ostree-remote-registry|ostree-image-signed|ostree-unverified-registry):.*$"
+var ostreeContainerTargetImgrefRegex = regexp.MustCompile("^(ostree-remote-registry|ostree-image-signed|ostree-unverified-registry):.*$")
 
 // Options for the org.osbuild.ostree.deploy.container stage.
 type OSTreeDeployContainerStageOptions struct {
@@ -29,7 +29,7 @@ type OSTreeDeployContainerStageOptions struct {
 func (OSTreeDeployContainerStageOptions) isStageOptions() {}
 
 func (options OSTreeDeployContainerStageOptions) validate() error {
-       exp := regexp.MustCompile(ostreeContainerTargetImgrefRegex)
+       exp := ostreeContainerTargetImgrefRegex
        if !exp.MatchString(options.TargetImgref) {
                return fmt.Errorf("'target_imgref' %q doesn't conform to schema (%s)", options.TargetImgref, exp.String())
        }
```
when the regex is invalid this yields:
```
$ go test ./pkg/osbuild/
panic: regexp: Compile(`(abc* ^(ostree-remote-registry|ostree-image-signed|ostree-unverified-registry):.*$`): error parsing regexp: missing closing ): `(abc* ^(ostree-remote-registry|ostree-image-signed|ostree-unverified-registry):.*$`

goroutine 1 [running]:
regexp.MustCompile({0xfdd886, 0x52})
	/usr/lib/go-1.21/src/regexp/regexp.go:319 +0xb4
github.com/osbuild/images/pkg/osbuild.init()
	/home/egon/devel/osbuild/images/pkg/osbuild/ostree_deploy_container_stage.go:8 +0x59
FAIL	github.com/osbuild/images/pkg/osbuild	0.020s
FAIL
```
the downside of course is an unnecessary compile if this function is never used so tradeoffs :)

Reply:
I commented on this in the previous PR here: https://github.com/osbuild/images/pull/243#discussion_r1404372500

I still think I prefer having the safety of a const.  And like you said there's the tiny addition of it only being compiled when needed.  I don't know if that outweighs catching it at test/compile time though.

Reply:
Thanks and sorry, I must have missed the reply last time. Fine of course